accepted/2021/statics-in-interfaces/README.md

```diff
@@ -8,6 +8,793 @@ C# is looking at enabling static abstract members in interfaces (https://github.
+        where TSelf : ISpanParseable<TSelf>
+    {
+        // This inherits from IParseable, much as ISpanFormattable is planned to inherit from IFormattable
+
+        static abstract TSelf Parse(ReadOnlySpan<char> s, IFormatProvider? provider);
+
+        static abstract bool TryParse(ReadOnlySpan<char> s, IFormatProvider? provider, out TSelf result);
+    }
+
+    public interface IEquatableOperators<TSelf, TOther> : IEquatable<TOther>
+        where TSelf : IEquatableOperators<TSelf, TOther>
+    {
+        // Given this takes two generic parameters, we could simply call it IEquatable<TSelf, TOther>
+
+        // This should not be `IEquatableOperators<TSelf>` as there are types like `Complex` where
+        // TOther can be `double` and represent an optimization, both in size and perf
+
+        static abstract bool operator ==(TSelf lhs, TOther rhs);
+
+        static abstract bool operator !=(TSelf lhs, TOther rhs);
+    }
+
+    public interface IComparableOperators<TSelf, TOther> : IComparable<TOther>, IEquatableOperators<TSelf, TOther>
+        where TSelf : IComparableOperators<TSelf, TOther>
+    {
+        // Given this takes two generic parameters, we could simply call it IComparable<TSelf, TOther>
+
+        // This inherits from IEquatableOperators<TSelf, TOther> as even though IComparable<T> does not
+        // inherit from IEquatable<T>, the <= and >= operators as well as CompareTo itself imply equality
+
+        static abstract bool operator <(TSelf lhs, TOther rhs);
+
+        static abstract bool operator <=(TSelf lhs, TOther rhs);
+
+        static abstract bool operator >(TSelf lhs, TOther rhs);
+
+        static abstract bool operator >=(TSelf lhs, TOther rhs);
+    }
+
+    public interface IAddable<TSelf, TOther>
+        where TSelf : IAddable<TSelf, TOther>
+    {
+        // An additive identity is not exposed here as it may be problematic for cases like
+        // DateTime that implements both IAddable<DateTime, DateTime> and IAddable<DateTime, TimeSpan>
+        // This would require overloading by return type and may be problematic for some scenarios where TOther might not have an additive identity
+
+        // We can't assume TOther + TSelf is valid as not everything is commutative
+
+        // We can't expose TSelf - TOther as some types only support one of the operations
+
+        static abstract TSelf operator +(TSelf value);
+
+        static abstract TSelf operator +(TSelf lhs, TOther rhs);
+
+        // There is an open question on if we should support "checked" versions of the operators
+        // Many algorithms, such as in LINQ, depend on checked behavior for overflow
+        // Supporting this would require well-defined ECMA-335 names and language work to ensure the right things happen
+    }
+
+    public interface ISubtractable<TSelf, TOther>
+        where TSelf : ISubtractable<TSelf, TOther>
+    {
+        static abstract TSelf operator -(TSelf lhs, TOther rhs);
```

Comment:
I like the idea of creating one interface per operator (or group of operators in the case of equality or comparison). It emphasizes the syntactic nature of the protocols versus building maintaining a hierarchy of algebraic structures (and any implied laws).